lib/ClangImporter/ImportDecl.cpp

```diff
@@ -2041,6 +2041,101 @@ namespace {
       fd->getAttrs().add(new (Impl.SwiftContext) UnsafeAttr(/*Implicit=*/true));
     }
 
+    void
+    generateStaticFactoryForCXXForeignRef(clang::ASTContext &clangCtx,
+                                          ClassDecl *result,
+                                          clang::CXXRecordDecl *cxxRecordDecl,
+                                          ClangImporter::Implementation &Impl) {
+
+      // Creating a name for the new synthesized function
+      clang::IdentifierTable &idents = clangCtx.Idents;
+      clang::IdentifierInfo *generatedFuncName =
+          &idents.get(("__returns" + cxxRecordDecl->getNameAsString()).c_str());
+
+      // Creating the return type for the synthesized function to be the
+      // pointe to cxxRecordDecl
+      const clang::Type *cxxRecordDeclTypePtr = cxxRecordDecl->getTypeForDecl();
+      clang::QualType cxxRecordTy(cxxRecordDeclTypePtr, 0);
+      clang::QualType cxxRecordPtrTy = clangCtx.getPointerType(cxxRecordTy);
+      // TODO/DOUBT: How to add `_Nonnull` as a qualifier on cxxRecordPtrTy
+      // return type ??
+
+      // Creating the type for the synthesized function
+      clang::FunctionProtoType::ExtProtoInfo EPI;
+      clang::QualType generatedFuncTy =
+          clangCtx.getFunctionType(cxxRecordPtrTy, {}, EPI);
+
+      clang::CXXMethodDecl *generatedCxxMethodDecl =
+          clang::CXXMethodDecl::Create(
+              clangCtx, cxxRecordDecl, cxxRecordDecl->getLocation(),
+              clang::DeclarationNameInfo(generatedFuncName,
+                                         clang::SourceLocation()),
+              generatedFuncTy, nullptr, clang::SC_Static, false, true,
+              clang::ConstexprSpecKind::Unspecified, clang::SourceLocation());
+
+      // Create and attach "returns_retained" attribute to synthesized method
+      clang::SwiftAttrAttr *returnsRetainedAttr =
+          clang::SwiftAttrAttr::Create(clangCtx, "returns_retained");
+      generatedCxxMethodDecl->addAttr(returnsRetainedAttr);
+
+      // Create and attach "swift_name("init()")" attribute to synthesized
+      // method
+      clang::SwiftNameAttr *swiftNameInitAttr =
+          clang::SwiftNameAttr::Create(clangCtx, "init()");
+      generatedCxxMethodDecl->addAttr(swiftNameInitAttr);
+
+      // Set the access specifier of the generated factory method to public
```

Comment:
Some of these comments are not really useful. E.g., in this particular case the comment is longer than the code itself and it does not add any information that is not there in the code already.